commit bf80e5d4259a192d6b06ae17e79a5e9dab48bf51
Author: Samuel Cabrero <scabrero@suse.de>
Date:   Mon Nov 30 19:02:51 2020 +0100

    cifs: Send witness register and unregister commands to userspace daemon
    
    + Define the generic netlink family commands and message attributes to
      communicate with the userspace daemon
    
    + The register and unregister commands are sent when connecting or
      disconnecting a tree. The witness registration keeps a pointer to
      the tcon and has the same lifetime.
    
    + Each registration has an id allocated by an IDR. This id is sent to the
      userspace daemon in the register command, and will be included in the
      notification messages from the userspace daemon to retrieve from the
      IDR the matching registration.
    
    + The authentication information is bundled in the register message.
      If kerberos is used the message just carries a flag.
    
    Signed-off-by: Samuel Cabrero <scabrero@suse.de>
    Reviewed-by: Aurelien Aptel <aaptel@suse.com>
    Signed-off-by: Steve French <stfrench@microsoft.com>

diff --git a/fs/cifs/cifs_swn.h b/fs/cifs/cifs_swn.h
new file mode 100644
index 000000000000..69c7bd1035da
--- /dev/null
+++ b/fs/cifs/cifs_swn.h
@@ -0,0 +1,17 @@
+/* SPDX-License-Identifier: GPL-2.0 */
+/*
+ * Witness Service client for CIFS
+ *
+ * Copyright (c) 2020 Samuel Cabrero <scabrero@suse.de>
+ */
+
+#ifndef _CIFS_SWN_H
+#define _CIFS_SWN_H
+
+struct cifs_tcon;
+
+extern int cifs_swn_register(struct cifs_tcon *tcon);
+
+extern int cifs_swn_unregister(struct cifs_tcon *tcon);
+
+#endif /* _CIFS_SWN_H */